### PR TITLE
Add compression library with test cases

### PR DESCRIPTION
**Pull request by NatashaWso2**

### CODE REVIEW

File: stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+     */
+    private static final int FOLDER_TO_UNZIP_INDEX = 1;
+
+    /**
+     * Decompress/unzip byte arrays/blob.
+     *  @param fileContentAsByteArray file content as a byte arry
+     * @param outputFolder           destination folder
+     * @param folderToUnzip folder to unzip in the zipped file
+     */
+    static void decompress(byte[] fileContentAsByteArray, String outputFolder, String folderToUnzip) {
```
Comment (manuranga) [inline stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java]: Shall change this to `static void decompress(InputStream zippedBytes, String outputFolder, String folderToUnzip)`

Then you will be able to unzip a file without reading it fully first.

File: stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    }
+
+    /**
+     * Extract files from the zipInputStream.
+     *
+     * @param in     zipInputStream object
+     * @param outdir output directory file
+     * @param name   name of the file
+     */
+    private static void extractFile(ZipInputStream in, Path outdir, String name) {
```
Comment (manuranga) [inline stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java]: See if you can use https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#copy(java.io.InputStream,%20java.nio.file.Path,%20java.nio.file.CopyOption...) instead

File: stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        return null;
+    }
+
+    /**
+     * Make directories if they doesn't exists.
+     *
+     * @param outdir destination file
+     * @param path   path of the destination directory
+     */
+    private static void mkdirs(Path outdir, String path) {
```
Comment (manuranga) [inline stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java]: can you reuse https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#createDirectories-java.nio.file.Path-java.nio.file.attribute.FileAttribute...-

File: stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java (hunk 1 above)
Comment (NatashaWso2) [reply stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java]: This method takes in a byte array of a compressed file and then unzips it.

Comment (NatashaWso2) [reply stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java]: Fixed with bc105af

File: stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java (hunk 2 above)
Comment (NatashaWso2) [reply stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java]: Fixed with bc105af

File: stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java (hunk 3 above)
Comment (NatashaWso2) [reply stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java]: Reused Files.createDirectories with bc105af

File: stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+    /**
+     * Create the directory name.
+     *
+     * @param name name of the directory
+     * @return directory name
+     */
+    private static String getDirectoryPath(String name) {
+        if (name != null) {
+            int s = name.lastIndexOf(File.separatorChar);
```
[the remark this answers is not in the document]
Comment (NatashaWso2) [reply stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java]: Checked with the .zip file specification it states that: 
4.4.17.1 The name of the file, with optional relative path. The path stored MUST not contain a drive or device letter, or a leading slash. All slashes MUST be forward slashes '/' as opposed to backwards slashes '\' for compatibility with Amiga and UNIX file systems etc. If input came from standard input, there is no file name field.

So changed it to "/" with ce6faad

**Pull request merged**